libswift/Sources/ExperimentalRegex/CMakeLists.txt

```diff
@@ -6,7 +6,11 @@
 # See http://swift.org/LICENSE.txt for license information
 # See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 
+file(GLOB_RECURSE LIBSWIFT_EXPERIMENTAL_REGEX_SOURCES
+  "${EXPERIMENTAL_STRING_PROCESSING_SOURCE_DIR}/Sources/_MatchingEngine/*.swift")
```

Comment:
Usually we don't collect sources with glob, but list all sources in the cmake files

Reply:
Because sources are from a separate repo that changes very rapidly, IMO listing sources would involve a lot of pull requests to apple/swift just to update the source list which then slows down development of the string processing repo. With glob, we can add files without submitting PRs the compiler repo unless we need to change the compiler interface.

Reply:
Either is ok, as we want to be tag-based. The hazard of manually listing files is that these things easily fall out of sync and we can think we've added API only to find out we haven't. We're not talking about just compiler-visible sources but adding public stdlib API, so this is a hazard that is not otherwise checked by the build system.

What's the hazard of globbing? I think this would be very interesting to know so that we're not cargo-culting an approach built for a completely different integration model.